### PR TITLE
[FIX] generateResourcesJson: Include dependencies of XML resources

### PR DESCRIPTION
**Pull request by matz3**
The XML analysis requires dependency resources to be able to list them
as "required" in the resources.json.

### CODE REVIEW

**Comment by matz3** [comment]
Tests are missing

File: lib/processors/resourceListCreator.js [hunk 1]
```diff
@@ -138,11 +138,13 @@ module.exports = async function({resources, options}) {
 		mergedResources: DEFAULT_BUNDLE_RESOURCES_FILTER,
 		designtimeResources: DEFAULT_DESIGNTIME_RESOURCES_FILTER,
 		supportResources: DEFAULT_SUPPORT_RESOURCES_FILTER,
-		debugBundles: DEBUG_BUNDLES
+		debugBundles: DEBUG_BUNDLES,
+		dependencyResources: []
```
Comment (RandomByte) [inline lib/processors/resourceListCreator.js:142]: Why make it part of the options and not of `parameters` itself (next to `resources`)?

**Review by RandomByte — CHANGES_REQUESTED** [review]
LGTM. Just one question regarding the function signature

File: lib/processors/resourceListCreator.js (hunk 1 above)
Comment (matz3) [reply lib/processors/resourceListCreator.js:142]: My idea was to keep the signature of having `resources` and `options` for processors, but it seems we already use other parameters in existing processors.
So having dependencyResources next to resources is probably better.

Comment (matz3) [reply lib/processors/resourceListCreator.js]: Done

**Comment by matz3** [comment]
CI is broken since https://github.com/SAP/ui5-builder/commit/61e2e3c43f52eaee7c51c1298c08a07ccc656058

**Comment by matz3** [comment]
CI should be solved via https://github.com/SAP/ui5-builder/pull/587

**Review by RandomByte — APPROVED** [review]

**Pull request merged**